docs/api/tensorflow_usage.rst

```diff
@@ -14,19 +14,32 @@ All configuration options are specified using environment variables as shown in
 Optimizing CPU Performance
 --------------------------
 
+.. note::
+    It is advisable to configure these options only if you are an advanced tensorflow user and understand
+    how different operations are implemented under the hood.
+
+These options are useful if you have implemented a custom component as part of your pipeline. Based on the tensorflow operations
+that the component uses, you can leverage multi-core CPU parallelism by tuning these options.
+
 Parallelizing One Operation
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Set ``TF_INTRA_OP_PARALLELISM_THREADS`` as an environment variable to specify the maximum number of threads that can be used
-to parallelize the execution of one operation. If left unspecified, this value defaults to ``0`` which means TensorFlow should
+to parallelize the execution of one operation. For example, operations like ``tf.matmul()`` and ``tf.reduce_sum`` can be executed
+on multiple threads running parallely. The default value for this variable is ``0`` which means TensorFlow should
 pick an appropriate value depending on the system configuration.
 
 Parallelizing Multiple Operations
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Set ``TF_INTER_OP_PARALLELISM_THREADS`` as an environment variable to specify the maximum number of threads that can be used
-to parallelize the execution of multiple **non-blocking** operations. If left unspecified, this value defaults to ``0``
-which means TensorFlow should pick an appropriate value depending on the system configuration.
+to parallelize the execution of multiple **non-blocking** operations. These would included operations which do not have a
+directed path between them in the tensorflow graph. The default value for this variable is ``0`` which means TensorFlow
```

Comment:
directed or direct?

Reply:
@akelad Should be directed because a direct path is something like A->B between two nodes A and B.
By directed I mean, consider A->B->D and C->D , then there is not directed path between A,C and B,C

Reply:
Added a small explanation